### PR TITLE
network: introduce log_neighbor_debug()

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/basic/ether-addr-util.c [hunk 1]
```diff
@@ -43,6 +43,22 @@ char* ether_addr_to_string(const struct ether_addr *addr, char buffer[ETHER_ADDR
         return buffer;
 }
 
+int ether_addr_to_string_alloc(const struct ether_addr *addr, char **ret) {
+        char *buf;
+
+        assert(addr);
+        assert(ret);
+
+        buf = malloc(ETHER_ADDR_TO_STRING_MAX);
```
Comment (poettering) [inline src/basic/ether-addr-util.c]: `new(char, ETHER_ADDR_TO_STRING_MAX)` would be a typesafe version

File: src/network/networkd-neighbor.c [hunk 2]
```diff
@@ -213,6 +213,28 @@ static bool neighbor_equal(const Neighbor *n1, const Neighbor *n2) {
         return neighbor_compare_func(n1, n2) == 0;
 }
 
+static void log_neighbor_debug(const Neighbor *neighbor, const char *str, const Link *link) {
+        assert(neighbor);
+        assert(str);
+
+        if (DEBUG_LOGGING) {
```
Comment (poettering) [inline src/network/networkd-neighbor.c]: I'd always invert the check here, and return here early, in order to keep indentation levels low. i.e.:

```c
if (!DEBUG_LOGGING)
        return;
```

but doesn't matter

**Review by poettering — APPROVED** [review]
minor stuff

**Comment by yuwata** [comment]
@poettering Thank you for the review. The above two points are addressed. Upgrading the green label.

**Comment by yuwata** [comment]
CentOS CI failure is #19463, not related. Merging.

**Pull request merged**